sklearn/metrics/_plot/tests/test_cap_curve_display.py

```diff
@@ -0,0 +1,537 @@
+            alpha=0.8,
+        )
+    else:
+        display = CAPCurveDisplay.from_predictions(
+            y,
+            y_pred,
+            pos_label=pos_label,
+            sample_weight=sample_weight,
+            alpha=0.8,
+        )
+
+    assert display.estimator_name == default_name, (
+        f"{display.estimator_name} != {default_name}"
+    )
+
+    cumulative_total = display.cumulative_total
+    y_true_cumulative = display.y_true_cumulative
+    assert np.all(y_true_cumulative >= cumulative_total - 1e-8), (
+        f"CAP curve dips below the chance line.\n"
+        f"x: {cumulative_total}\n"
+        f"y: {y_true_cumulative}"
+    )
+
+    perfect_y = np.interp(
+        display.cumulative_total, display._perfect_x, display._perfect_y
+    )
+    assert np.all(y_true_cumulative <= perfect_y + 1e-8), (
+        "CAP curve rises above the perfect prediction line.\n"
+        f"x: {perfect_y}\n"
+        f"y: {y_true_cumulative}"
+    )
+
+
+def test_lorenz_curve_valid_position(pyplot):
+    estimator = PoissonRegressor()
+
+    X, y = make_regression(n_samples=100, n_features=2, noise=10.0, random_state=42)
+    y = y - y.min() + 1e-3
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, y, test_size=0.4, random_state=42
+    )
+    estimator.fit(X_train, y_train)
+    y_pred = estimator.predict(X_test)
+
+    display = CAPCurveDisplay.from_predictions(y_test, y_pred)
+
+    cumulative_total = display.cumulative_total
+    y_true_cumulative = display.y_true_cumulative
+    assert np.all(cumulative_total >= y_true_cumulative - 1e-8), (
+        f"Lorenz curve dips above the chance line.\n"
+        f"x: {cumulative_total}\n"
+        f"y: {y_true_cumulative}"
+    )
+
+    perfect_y = np.interp(
+        display.cumulative_total, display._perfect_x, display._perfect_y
+    )
+    assert np.all(y_true_cumulative >= perfect_y - 1e-8), (
+        "Lorenz curve rises below the perfect prediction line.\n"
+        f"x: {perfect_y}\n"
+        f"y: {y_true_cumulative}"
+    )
+
+
+@pytest.mark.parametrize(
+    "response_method", ["predict_proba", "decision_function", "auto"]
+)
+def test_cap_for_non_prob_classifier(pyplot, data_binary, response_method):
+    X, y = data_binary
+    svc = LinearSVC().fit(X, y)
+
+    if response_method == "predict_proba":
+        match = "LinearSVC has none of the following attributes: predict_proba."
+        with pytest.raises(
+            AttributeError,
+            match=match,
+        ):
+            CAPCurveDisplay.from_estimator(svc, X, y, response_method=response_method)
+    else:
+        CAPCurveDisplay.from_estimator(svc, X, y, response_method=response_method)
+
+
+def test_y_true_with_negative_values(pyplot):
+    match = "`y_true` contains negative values, which isn't allowed for "
+    "continuous targets. If your data shouldn't be treated as "
+    "continuous, try converting the values to integers or strings "
+    "instead."
```

Comment:
I think this code is not testing what it's supposed to test: to define a multiline string you need parens, otherwise only the first line is assigned to the `match` variable and the rest is treated as a comment in Python.

```suggestion
    match = (
        "`y_true` contains negative values, which isn't allowed for "
        "continuous targets. If your data shouldn't be treated as "
        "continuous, try converting the values to integers or strings "
        "instead."
    )
```

Reply:
You might also need to `re.escape(...)` around that string to avoid getting the punctuation interpreted as regular expression operators.